src/rviz/default_plugin/marker_utils.cpp

```diff
@@ -76,4 +84,346 @@ MarkerBase* createMarker(int marker_type, MarkerDisplay* owner, DisplayContext*
+{
+  if(marker.points.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "points should not be empty for specified marker type" ;
+  }
+  else if(marker.type == visualization_msgs::Marker::TRIANGLE_LIST && (marker.points.size() % 3) != 0)
+  {
+    addCommaIfRequired(ss);
+    ss << "number of points should be a multiple of 3 for TRIANGLE_LIST Marker";
+  }
+  else if(marker.type == visualization_msgs::Marker::LINE_LIST && (marker.points.size() % 2) != 0)
+  {
+    addCommaIfRequired(ss);
+    ss << "number of points should be a multiple of 2 for LINE_LIST Marker";
+  }
+  else if(marker.type == visualization_msgs::Marker::LINE_STRIP && marker.points.size() <= 1)
+  {
+    addCommaIfRequired(ss);
+    ss << "at least two points are required for a LINE_STRIP Marker";
+  }
+}
+
+void checkPointsEmpty(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(!marker.points.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "points array is ignored by specified marker type";
+  }
+}
+
+void checkColors(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.colors.size() == 0)
+  {
+    checkColor(marker, ss);
+  }
+
+  if(marker.colors.size() != marker.points.size())
+  {
+    addCommaIfRequired(ss);
+    ss << "number of colors is not equal to number of points or 0";
+  }
+}
+
+void checkColorsEmpty(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(!marker.colors.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "colors array is ignored by specified marker type";
+  }
+}
+
+void checkTextNotEmptyOrWhitespace(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.text.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "text is empty for TEXT_VIEW_FACING type marker";
+  }
+  else if(marker.text.find_first_not_of(" \t\n\v\f\r") == std::string::npos)
+  {
+    addCommaIfRequired(ss);
+    ss << "text of TEXT_VIEW_FACING Marker only consists of whitespaces";
+  }
+}
+
+void checkTextEmpty(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(!marker.text.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "text is ignored for specified marker type";
+  }
+}
+
+void checkMesh(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.mesh_resource.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "path to mesh resource is empty for MESH_RESOURCE marker";
+  }
+}
+
+void checkMeshEmpty(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if (!marker.mesh_resource.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "mesh_resource is ignored for specified marker type";
+  }
+  if (marker.mesh_use_embedded_materials)
+  {
+    addCommaIfRequired(ss);
+    ss << "using embedded materials is not supported for markers other than MESH_RESOURCE";
+  }
+}
+
+void addCommaIfRequired(std::stringstream& ss)
+{
+  if (ss.tellp() != 0) // check if string is not empty
+  {
+    ss << ", ";
```

Comment:
What about using newline instead of comma?
The list of warnings might get quite long...

Reply:
I think that is a good idea since the error messages might get quite long and the different errors are more clearly separated by a new line. Implemented in c028953

Reply:
In addition to a newline, a comma is nice to have for the rviz status display since newlines are not displayed there. I did this in 3a75c5b